### DISCUSSION TITLE
Is there a way to configure flags set when pushing changes

### DISCUSSION THREAD

**Discussion by plejd-sebman** [jesseduffield/lazygit#3582] [Ideas]
- more specifically, I want to be able to provide special force push flag called `--force-with-lease` as documented here. https://git-scm.com/docs/git-push#Documentation/git-push.txt---no-force-with-lease
- would that be possible?

**Comment by stefanhaller** [comment]
Lazygit already provides `--force-with-lease` for you, as you can see in the command log in the lower right corner of the lazygit screen (type `@` to show it if you don't see it).

Let me warn you though (as I always do when this topic comes up) that `--force-with-lease` on its own doesn't offer the protection that most people think it does. There's an unfortunate amount of misinformation about it out there on the Internet, making it sound like `--force-with-lease` is a safer variant of `--force`, protecting you from accidentally overwriting other people's commits. That's simply not true for most people, especially for lazygit users that have auto-fetch enabled by default.

What you want to do is turn on `--force-if-includes` in addition to `--force-with-lease`. You can do this globally by running `git config --global push.useForceIfIncludes true`.

**Reply by plejd-sebman** [reply]
@stefanhaller, many thanks for taking the time to answer my question. I've configured it accordingly and disabled `auto-fetch`.

**Reply by stefanhaller** [reply]
You don't need to disable auto-fetch fetch if you turn on `useForceIfIncludes`. You'll still be safe.
